### DISCUSSION TITLE
Is is possible for a keybinding to run multiple commands?

### DISCUSSION THREAD

**Discussion by joshgoebel** [zed-industries/zed#14422] [Feature Requests]
For example I'd like `j k l` to exit from insert to normal mode, and then immediately also save the file.

**Comment by 0x2CA** [comment]
There is currently no way to key-bind multiple commands, but you can key-bind each command individually and then use the simulated key method to achieve the.
for example:
```
[
  {
    "context": "Editor  && vim_mode == insert && !VimWaiting && !menu",
    "bindings": {
      "j k l": ["workspace::SendKeystrokes", "escape : w enter"]
    }
  }
]
```

**Comment by FiedorPw** [comment]
Thank you! my bind for space in normal mode works just fine 
```json
 {
    // Bind for using space to insert a space in normal mode
    "context": "Editor && vim_mode == normal",
    "bindings": {
      "space": ["workspace::SendKeystrokes", "i space escape"]
    }
```
  }

**Comment by axelson** [comment]
Running multiple commands is actually possible now via `action::Sequence`: https://github.com/zed-industries/zed/pull/39261

Although it doesn't include any special handling for async commands. So I'm not sure if it would work for your specific scenario.

**Reply by aurexav** [reply]
This should be updated to doc.

**Reply by krzkaczor** [reply]
@axelson super cool! Any plans to support async actions or arbitrary sleeps?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
